src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int RADIUS = 10;
+    private static final int MAX_NUM = 20;
+
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+    private final Figure circle = new Circle(colorSupplier.getRandomColor(),
```

Comment:
It's better to create random figures in the `getRandomFigure` method rather than initializing them as instance variables. This will allow you to generate different figures each time the method is called.